src/contracts/token/Eigen.sol

```diff
@@ -123,17 +123,22 @@ contract Eigen is OwnableUpgradeable, ERC20VotesUpgradeable {
      */
     function wrap(uint256 amount) external {
         require(bEIGEN.transferFrom(msg.sender, address(this), amount), "Eigen.wrap: bEIGEN transfer failed");
-        _transfer(address(this), msg.sender, amount);
+        _mint(msg.sender, amount);
     }
 
     /**
      * @notice This function allows Eigen holders to unwrap their tokens into bEIGEN
      */
     function unwrap(uint256 amount) external {
-        _transfer(msg.sender, address(this), amount);
+        _burn(msg.sender, amount);
         require(bEIGEN.transfer(msg.sender, amount), "Eigen.unwrap: bEIGEN transfer failed");
     }
 
+    // @notice Burns EIGEN tokens held by the EIGEN token address itself
+    function burnExtraTokens() external {
+        _burn(address(this), balanceOf(address(this)));
```

Comment:
Should this instead be
```solidity
uint256 unbackedTokens =  totalSupply() - bEIGEN.balanceOf(address(this));
require(unbackedTokens != 0, "Eigen.burnExtraTokens: No unbacked tokens to burn");
_burn(address(this), unbackedTokens);
```
essentially burning all tokens that are not backed which should only work one time for the tokens living at the Eigen token address.

My main goal with this would be so that we don't fuck up invariants when someone stupidly sends Eigen tokens directly to the Eigen token contract.

Reply:
discussed offline; we decided to remove this function and go another route for fixing total supply, see https://github.com/Layr-Labs/eigenlayer-contracts/pull/717/commits/07c1fa6c7f9c4a631d9fa4f89a80f0526e0838c9